File: Rust/ExampleTests/common/cartinput_typed.rs
#![allow(dead_code, unused_imports, unused_variables)]

use super::json;
use super::cartinput_string::CartInputString;

#[derive(Debug, Clone, Default, PartialEq)]
pub struct CartInputTyped {
    pub totalitems: String,
    pub shipping: String,
    pub discount: String,
    pub total_price: String,
    pub notes: String,
}

impl CartInputTyped {
    pub fn from_str_struct(s: &CartInputString) -> Self {
        Self {
            totalitems: s.totalitems.clone(),
            shipping: s.shipping.clone(),
            discount: s.discount.clone(),
            total_price: s.total_price.clone(),
            notes: s.notes.clone(),
        }
    }

    pub fn to_json_value(&self) -> json::Value {
        json::Value::Object(vec![
            ("totalitems".to_string(), json::Value::Str(self.totalitems.clone())),
            ("shipping".to_string(), json::Value::Str(self.shipping.clone())),
            ("discount".to_string(), json::Value::Str(self.discount.clone())),
            ("total_price".to_string(), json::Value::Str(self.total_price.clone())),
            ("notes".to_string(), json::Value::Str(self.notes.clone())),
        ])
    }

    pub fn to_json(&self) -> String {
        json::write(&self.to_json_value())
    }

    pub fn from_json_value(v: &json::Value) -> json::JsonResult<Self> {
        Ok(Self {
            totalitems: json::as_string(json::require(v, "totalitems")?, "totalitems")?,
            shipping: json::as_string(json::require(v, "shipping")?, "shipping")?,
            discount: json::as_string(json::require(v, "discount")?, "discount")?,
            total_price: json::as_string(json::require(v, "total_price")?, "total_price")?,
            notes: json::as_string(json::require(v, "notes")?, "notes")?,
        })
    }

    pub fn from_json(text: &str) -> json::JsonResult<Self> {
        Self::from_json_value(&json::parse(text)?)
    }

    pub fn to_json_list(list: &[CartInputTyped]) -> String {
        json::write(&json::Value::Array(
            list.iter().map(|item| item.to_json_value()).collect(),
        ))
    }

    pub fn from_json_list(text: &str) -> json::JsonResult<Vec<CartInputTyped>> {
        let root = json::parse(text)?;
        let items = json::as_array(&root, "CartInputTyped")?;
        let mut result = Vec::with_capacity(items.len());
        for e in items {
            result.push(Self::from_json_value(e)?);
        }
        Ok(result)
    }
}

File: Swift/ExampleTests/common/CartInputTyped.swift
public struct CartInputTyped: Equatable, CustomStringConvertible {
    public let totalItems: String
    public let shipping: String
    public let discount: String
    public let totalPrice: String
    public let notes: String

    public init(totalItems: String, shipping: String, discount: String, totalPrice: String, notes: String) {
        self.totalItems = totalItems
        self.shipping = shipping
        self.discount = discount
        self.totalPrice = totalPrice
        self.notes = notes
    }

    public init(from s: CartInputString) {
        self.totalItems = s.totalItems
        self.shipping = s.shipping
        self.discount = s.discount
        self.totalPrice = s.totalPrice
        self.notes = s.notes
    }

    public func toStringStruct() -> CartInputString {
        return CartInputString(
            totalItems: String(describing: totalItems),
            shipping: String(describing: shipping),
            discount: String(describing: discount),
            totalPrice: String(describing: totalPrice),
            notes: String(describing: notes)
        )
    }

    public static func toStringList(_ list: [CartInputTyped]) -> [CartInputString] {
        return list.map { $0.toStringStruct() }
    }

    public static func fromStringList(_ list: [CartInputString]) -> [CartInputTyped] {
        return list.map { CartInputTyped(from: $0) }
    }

    public func toJSONValue() -> [String: Any] {
        return [
            "totalItems": totalItems,
            "shipping": shipping,
            "discount": discount,
            "totalPrice": totalPrice,
            "notes": notes,
        ]
    }

    public func toJSON() throws -> String {
        return try Json.write(toJSONValue())
    }

    public init(fromJSONValue m: [String: Any]) throws {
        self.totalItems = try Json.asString(Json.require(m, "totalItems"), "totalItems")
        self.shipping = try Json.asString(Json.require(m, "shipping"), "shipping")
        self.discount = try Json.asString(Json.require(m, "discount"), "discount")
        self.totalPrice = try Json.asString(Json.require(m, "totalPrice"), "totalPrice")
        self.notes = try Json.asString(Json.require(m, "notes"), "notes")
    }

    public init(fromJSON text: String) throws {
        try self.init(fromJSONValue: Json.parseObject(text))
    }

    public static func toJSONList(_ list: [CartInputTyped]) throws -> String {
        return try Json.write(list.map { $0.toJSONValue() })
    }

    public static func fromJSONList(_ text: String) throws -> [CartInputTyped] {
        return try Json.parseArray(text).map {
            try CartInputTyped(fromJSONValue: Json.asObject($0, "CartInputTyped"))
        }
    }

    public var description: String {
        return "TotalItems=\(totalItems), Shipping=\(shipping), Discount=\(discount), Total Price=\(totalPrice), Notes=\(notes)"
    }
}
